metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py

```diff
@@ -216,14 +222,10 @@ def get_bigquery_profile_request(
         if not self.is_dataset_eligible_for_profiling(
             dataset_name, table.last_altered, table.size_in_bytes, table.rows_count
         ):
-            # Profile only table level if dataset is filtered from profiling
-            # due to size limits alone
-            if self.is_dataset_eligible_for_profiling(
-                dataset_name, table.last_altered, 0, 0
-            ):
-                profile_table_level_only = True
-            else:
-                skip_profiling = True
+            profile_table_level_only = True
+            self.report.num_tables_not_eligible_profiling[dataset] = (
```

Comment:
suggestion: make `num_tables_not_eligible_profiling` a Counter and no need to do this tricks.
just: `self.report.num_tables_not_eligible_profiling[dataset]++`
https://docs.python.org/3/library/collections.html

Reply:
we try to use in report structs that can't grow indefinitely and there this way I think won't work :(